### PR TITLE
EP-7110: Refactor HGNCMultipleGenes to allow comparison with threshold

### PR DESCRIPTION
**Pull request by luca-drf**
Refactor `HGNCMultipleGenes` into `CompareHGNCMultipleGenes`.

The new DC will fail only if the increase in HGNC symbols assigned to multiple genes compared to the previous release is above 5%.

### CODE REVIEW

File: lib/Bio/EnsEMBL/DataCheck/Checks/CompareHGNCMultipleGenes.pm [hunk 1]
```diff
@@ -16,45 +16,65 @@ limitations under the License.
...
+  my $cur_multi_hgnc_percentage = $self->multiple_hgnc_percentage($cur_dba);
+  my $old_multi_hgnc_percentage = 0;
+
+  if (defined $old_dba) {
+    $old_multi_hgnc_percentage = $self->multiple_hgnc_percentage($old_dba);
+  }
+
+  my $multi_hgnc_diff = $cur_multi_hgnc_percentage - $old_multi_hgnc_percentage;
+
+  my $desc = "HGNC symbols assigned to multiple genes have not increased more than 5%";
```
Comment (marcoooo) [inline lib/Bio/EnsEMBL/DataCheck/Checks/CompareHGNCMultipleGenes.pm:53]: Do we expect HGNC symbol to increase by more than 5% every release?

Comment (luca-drf) [reply lib/Bio/EnsEMBL/DataCheck/Checks/CompareHGNCMultipleGenes.pm:53]: Genebuild suggested to fail the test when the proportion of HGNC symbols assigned to multiple genes increases more than 5% from the previous release.

Comment (marcoooo) [reply lib/Bio/EnsEMBL/DataCheck/Checks/CompareHGNCMultipleGenes.pm:53]: Was just wondering if a drop had the same potential issue.

**Review by marcoooo — APPROVED** [review]

File: lib/Bio/EnsEMBL/DataCheck/Checks/CompareHGNCMultipleGenes.pm (hunk 1 above)
Comment (luca-drf) [reply lib/Bio/EnsEMBL/DataCheck/Checks/CompareHGNCMultipleGenes.pm:53]: I did wonder the same thing, but I'm guessing it's not a problem to have several less symbols assigned to multiple genes. Also, this a very specific test and we have several other checks in place for HGNC and xrefs, so it's probably okay. I think we can ping Genebuild again and tune the threshold or add another check if we find that this one is too permissive.

**Pull request merged**